core/src/main/scala/org/apache/spark/sql/streaming/eventhubs/EventHubsSourceProvider.scala

```diff
@@ -43,50 +46,56 @@ private[sql] class EventHubsSourceProvider
                             schema: Option[StructType],
                             providerName: String,
                             parameters: Map[String, String]): Source = {
-    // TODO: use serviceLoader EH client dependency injection
     EventHubsClientWrapper.userAgent =
       s"Structured-Streaming-${sqlContext.sparkSession.sparkContext.version}"
-    new EventHubsSource(sqlContext, parameters, EventHubsClientWrapper.apply)
+
+    val caseInsensitiveParameters = parameters.map {
+      case (k, v) => (k.toLowerCase(Locale.ROOT), v)
+    }
+
+    new EventHubsSource(sqlContext,
+                        parameters,
+                        EventHubsClientWrapper.apply,
+                        metadataPath,
+                        failOnDataLoss(caseInsensitiveParameters))
   }
+
+  private def failOnDataLoss(caseInsensitiveParams: Map[String, String]) =
+    caseInsensitiveParams.getOrElse(FailOnDataLossOptionKey, "true").toBoolean
 }
 
 private[sql] object EventHubsSourceProvider extends Serializable {
+  private val FailOnDataLossOptionKey = "eventhubs.failondataloss"
 
-  private[eventhubs] def ifContainsPropertiesAndUserDefinedKeys(
-      parameters: Map[String, String]): (Boolean, Seq[String]) = {
+  def sourceSchema(parameters: Map[String, String]): StructType = {
     val containsProperties =
       parameters.getOrElse("eventhubs.sql.containsProperties", "false").toBoolean
+
     val userDefinedKeys = {
       if (parameters.contains("eventhubs.sql.userDefinedKeys")) {
```

Comment:
Is there plans to open up typing for custom properties?

Reply:
Not explicitly. Once this re-write is complete, we'll start integration testing and planning nice things to layer on top of the new library. I will definitely contact you for input around then!